x-pack/heartbeat/monitors/browser/synthexec/execmultiplexer.go

```diff
@@ -28,20 +28,20 @@ func (e ExecMultiplexer) writeSynthEvent(se *SynthEvent) {
 		e.currentJourney.Store(true)
 		e.eventCounter.Store(-1)
 	}
-	hasCurrentJourney := e.currentJourney.Load()
+	se.index = e.eventCounter.Inc()
+
 	if se.Type == "journey/end" || se.Type == "cmd/status" {
 		e.currentJourney.Store(false)
 	}
-
-	se.index = e.eventCounter.Inc()
-	if hasCurrentJourney {
-		e.synthEvents <- se
-	}
+	e.synthEvents <- se
 }
 
 // SynthEvents returns a read only channel for synth events
-func (e ExecMultiplexer) SynthEvents() <-chan *SynthEvent {
-	return e.synthEvents
+func (e ExecMultiplexer) SynthEvents(inline bool) <-chan *SynthEvent {
+	if inline || e.currentJourney.Load() {
+		return e.synthEvents
+	}
+	return make(chan *SynthEvent)
```

Comment:
I don't quite get this. Why do we return an empty channel here that will never be closed? If this has a purpose it could use a comment explaining why it's necessary.

My reading of this is it's a race, and if there's not a running journey at the time `SynthEvents(false)` is invoked for a suite (i.e. it's invoked before or between suites) the caller will get nothing back.

Reply:
This code should go away, Missed to remove this before the review.